selftests/unit/test_utils_network.py

```diff
@@ -0,0 +1,37 @@
+import unittest
+
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+from avocado.utils import network
+
+
+class PortTrackerTest(unittest.TestCase):
+
+    def test_register_port(self):
+        tracker = network.PortTracker()
+        network.is_port_free = mock.MagicMock(return_value=True)
+        self.assertNotIn(22, tracker.retained_ports)
+        tracker.register_port(22)
+        network.is_port_free.assert_called_once_with(22, tracker.address)
+        self.assertIn(22, tracker.retained_ports)
+
+    def test_release_port_does_not_poke_system(self):
+        tracker = network.PortTracker()
+        tracker.release_port = mock.MagicMock()
+        network.is_port_free = mock.MagicMock()
+        tracker.release_port(22)
+        tracker.release_port.assert_called_once_with(22)
+        network.is_port_free.assert_not_called()
```

Comment:
This test is IMO unnecessary as it basically verifies that mock was not mocked (I know there is some value, but complete test which registers port and releases it would cover the same functionality and would work the same even when we modify the port tracker.

Reply:
Not really unnecessary IMO. It tests the behavior I expect from it, that is, there's no need to check if a port is free (or not) to release it.  This should be done by the user of the PortTracker.